synapse/handlers/sync.py

```diff
@@ -794,7 +797,7 @@ def compute_state_delta(self, room_id, batch, sync_config, since_token, now_toke
             else:
                 state_ids = {}
                 if lazy_load_members:
-                    if types:
+                    if types and batch.events:
```

Comment:
so this can go back to being just `types` now, right?

Reply:
it seems safer to keep it there, as a cautionary bounds check? (e.g. if/when we enable LL for incr syncs again)

Reply:
the problem is really that the belt-and-braces check on `types` *and* `batch.events` makes me wonder what I am missing.

I guess you can argue this both ways though, so \<shrug>